src/main/java/io/pinecone/clients/Pinecone.java

```diff
@@ -149,14 +152,27 @@ public void deleteCollection(String collectionName) throws PineconeException {
         }
     }
 
-    public Index createIndexConnection(String indexName) {
-        PineconeConnection connection = new PineconeConnection(config, indexName);
-        return new Index(connection);
+    public Index getIndexConnection(String indexName) {
+        config.setHost(getIndexHost(indexName));
+        PineconeConnection connection = getConnection(indexName);
+        return new Index(this, connection, indexName);
+    }
+
+    public AsyncIndex getAsyncIndexConnection(String indexName) {
+        PineconeConnection connection = getConnection(indexName);
+        return new AsyncIndex(this, connection, indexName);
+    }
```

Comment:
Do we not need to do the host lookup in this method?

Reply:
Good catch!!